rfcs/0000-policy_target_scope.md

```diff
@@ -0,0 +1,228 @@
+
+# Motivation
+[motivation]: #motivation
+
+As per the specification, a given resource will always need to support multiple policies targeting it.
+Even if we only support a single e.g. `RateLimitPolicy` per resource, there will alway be a
+possibility for another policy being defined else where in that resource's hierarchy: e.g. _rlp1_
+targeting the `Gateway`, while _rlp2_ targeting the `HTTPRoute` being attached to the `Gateway`. Until
+there is an `HTTPRoute` tho, _rlp1_ will have no semantic, as there is no traffic to rate limit. When
+an `HTTPRoute` is attached to the `Gateway` is the point when that _rlp1_ becomes applicable. If that
+`HTTPRoute` also defines a `RateLimitPolicy`, the two get merged, according to the `default` &
+`override` rules of the specification. 
+
+The idea is to keep things simple (if only for now), so that when traffic is routed through an
+`HTTPRoute` all applicable policies of a `kind` have been merged into a single one _and_ that single
+policy then applies to the whole traffic of that route.
+
+The current approach, which lets each policy define an arbitrary subset of the traffic to which it
+applies, makes it impossible to know how to merge the different policies into one. Each policy's set
+requires to know how to applicable it becomes to the union and/or the intersection of the said
+subsets. While a heuristic could be applied, there are use-cases where the other option will be
+desirable. The question then becomes which policy is the authority as to decide which it should be. In
+order to avoid having to answer these questions, this proposal looks at not letting a policy
+discriminate which part of traffic it applies to: it's the whole traffic, always.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Traffic to your service is ultimately routed when an `HTTPRoute` binds at least one  `Gateway`, thru
+the
+[`parentRefs`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.ParentReference),
+and wires traffic to the
+[`backendRef`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPBackendRef),
+i.e. your service, according to the
+[`HTTPRouteMatch`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteMatch)
+of a
+[`HTTPRouteRule`](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteRule)
+define on that `HTTPRoute`. All the traffic matched by _any_ `HTTPRouteRule` of a given `HTTPRoute`
+will be affected by which ever `RateLimitPolicy` or `AuthPolicy` you attach to it.
+
+Any Kuadrant `Policy` will apply to the entirety of the traffic that is routed through the `HTTPRoute`
+it is eventually attached to. A `Policy` can be attached directly to an `HTTPRoute`. It will apply its
+specification according to the `override` and `default` section of the `Policy`'s `spec`. Let's use a
+simple example:
+
+```yaml
```

Comment:
so I like the simplicity being aimed at here but I worry we will need to allow some form of conditional here
example if we were to always send common descriptors (if we agree they are common) e.g Headers. Would we see a way for someone to define 

```
limits:
        - conditions:
            - "req.path=='/checkout'"
            - "req.method=='POST'"
          maxValue: 5
          seconds: 10
          variables: []
```

Reply:
But I feel you are aiming at something very different. I had thought perhaps by applying a policy to all traffic, it had meant all traffic would be sent to the implementing service and then the implementing service could make a conditional decision